spec/outputs/file_spec.rb

```diff
@@ -418,5 +418,37 @@
         end
       end
     end
+
+    context "with non-zero flush interval" do
+      let(:tmp) { Stud::Temporary.pathname }
+
+      let(:events) do
+        100.times.map do |idx|
+          LogStash::Event.new("value" => idx)
+        end
+      end
+
+      let(:config) {
+        {
+            "path" => tmp,
+            "codec" => LogStash::Codecs::JSONLines.new,
+            "flush_interval" => 1
+        }
+      }
+      let(:output) { LogStash::Outputs::File.new(config) }
+
+      before(:each) { output.register }
+      after(:each) { File.exist?(tmp) && File.unlink(tmp) }
+
+      it 'flushes without requiring additional events' do
+        output.multi_receive(events)
+        expect(File.size(tmp)).to eq(0)
```

Comment:
This is perhaps too strict, and is causing CI to fail on older JRuby which auto-flushes the underlying file descriptors: https://travis-ci.org/logstash-plugins/logstash-output-file/jobs/397140556#L1045